### PR TITLE
feat(exchange): compare chart type

### PR DESCRIPTION
**Pull request by vbstreetz**
- Implements new chart type
- Moves some more chart logic into hooks

### CODE REVIEW

File: sections/exchange/TradeCard/Charts/PriceChartCard.tsx [hunk 1]
```diff
@@ -145,134 +144,63 @@ const ChartCard: FC<ChartCardProps> = ({
...
+									isActive={period.period === selectedPeriod}
+									onClick={(event) => {
+										setSelectedPeriod(period.period);
+									}}
+								>
+									{t(period.i18nLabel)}
+								</StyledTextButton>
+							))}
+						</PeriodSelector>
+						{!(!isSUSD && selectedPeriod === Period.ONE_MONTH) ? null : (
```
Comment (JChiaramonte7) [inline sections/exchange/TradeCard/Charts/PriceChartCard.tsx:164]: So you might have to update to the latest candlesticks from dev. But the current logic is to **always** show the candlestick option, but put the user on 1M time period if they click it. Which is something we should carry over here.

File: sections/exchange/TradeCard/Charts/CombinedPriceChartCard.tsx [hunk 2]
```diff
@@ -74,31 +99,45 @@ const ChartCard: FC<ChartCardProps> = ({
...
+		(isCandleStickChart && noCandleSticksChartData) || (isAreaChart && noAreaChartData);
 
-	const fontStyle = {
-		fontSize: '12px',
-		fill: theme.colors.white,
-		fontFamily: theme.fonts.mono,
-	};
+	useEffect(() => {
+		if (selectedChartType === ChartType.CANDLESTICK && selectedPeriod !== Period.ONE_MONTH) {
+			// candlesticks type is only available on monthly view
```
Comment (JChiaramonte7) [inline sections/exchange/TradeCard/Charts/CombinedPriceChartCard.tsx:127]: 👍

File: sections/exchange/TradeCard/Charts/CombinedPriceChartCard.tsx [hunk 3]
```diff
@@ -125,115 +164,90 @@ const ChartCard: FC<ChartCardProps> = ({
...
+								<CompareRatioToggleType
+									onClick={() => {
+										setSelectedChartType(ChartType.AREA);
+									}}
+									isActive={isAreaChart}
+								>
+									{t('common.chart-types.ratio')}
+								</CompareRatioToggleType>
+								{selectedPeriod !== Period.ONE_MONTH ? null : (
+									<CompareRatioToggleType
```
Comment (JChiaramonte7) [inline sections/exchange/TradeCard/Charts/CombinedPriceChartCard.tsx:192]: I think you should refactor `ChartTypeToggle` to use this text version of the toggle and use it on both the Double and Single view. That way we are more consistent.

**Review by JChiaramonte7 — CHANGES_REQUESTED** [review]
This is looking good @vbstreetz. Just a few more tweaks to fix logic and clean things up.

File: sections/exchange/TradeCard/Charts/PriceChartCard.tsx (hunk 1 above)
Comment (vbstreetz) [reply sections/exchange/TradeCard/Charts/PriceChartCard.tsx]: Gotcha. Fixed.

File: sections/exchange/TradeCard/Charts/CombinedPriceChartCard.tsx (hunk 3 above)
Comment (vbstreetz) [reply sections/exchange/TradeCard/Charts/CombinedPriceChartCard.tsx]: Fixed.

File: sections/exchange/TradeCard/Charts/CombinedPriceChartCard.tsx [hunk 4]
```diff
@@ -207,9 +217,12 @@ const CombinedPriceChartCard: FC<CombinedPriceChartCardProps> = ({
 							{PERIOD_LABELS.map((period) => (
 								<StyledTextButton
 									key={period.period}
-									isActive={period.period === selectedPeriod}
+									isActive={period.period === selectedChartPeriod}
+									disabled={
+										period.period !== Period.ONE_MONTH && isCandleStickChart && isOneMonthPeriod
```
Comment (JChiaramonte7) [inline sections/exchange/TradeCard/Charts/CombinedPriceChartCard.tsx:222]: I don't think we should disable the other time periods. We should automatically switch the user back to the area chart when they click on any period other than 1M. 

We don't want people to get stuck and complain.

**Review by JChiaramonte7 — APPROVED** [review]

File: sections/exchange/TradeCard/Charts/CombinedPriceChartCard.tsx (hunk 4 above)
Comment (vbstreetz) [reply sections/exchange/TradeCard/Charts/CombinedPriceChartCard.tsx]: Fixed.

**Pull request merged**